src/generator/models.ts

```diff
@@ -25,7 +25,8 @@ export async function generateModels(session: Session<CodeModel>): Promise<model
   let modelText = await contentPreamble(session);
 
   // we do model generation first as it can add imports to the imports list
-  const structs = generateStructs(modelImports, serdeImports, session.model.schemas.objects);
+  const omitXMsExternal = await session.getValue('omit-x-ms-external', true);
```

Comment:
I would prefer that the default is `false`, and that if `azure-arm` is `true` then we can default this to `true`.